demos/source/aws_iot_demo_jobs.c

```diff
@@ -0,0 +1,591 @@
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file aws_iot_demo_jobs.c
+ * @brief Demonstrates use of the AWS IoT Jobs library.
+ */
+
+/* Config include. Should always come first per the style guide. */
+#include "iot_config.h"
+
+/* System Includes */
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+/* Demo logging. */
+#include "iot_demo_logging.h"
+
+/* Platform Includes. */
+#include "platform/iot_clock.h"
+#include "platform/iot_network.h"
+#include "platform/iot_threads.h"
+
+/* MQTT Include. The underlying protocol for this demo. */
+#include "iot_mqtt.h"
+
+/* JSON Utilies for pulling out values. */
+#include "iot_json_utils.h"
+
+/* Jobs Library. */
+#include "aws_iot_jobs.h"
+
+/**
+ * @brief The Library Timeout.
+ */
+#define TIMEOUT_MS                      ( 5000u )
+
+/**
+ * @brief The MQTT Library keepalive time.
+ */
+#define KEEP_ALIVE_SECONDS              ( 60 )
+
+/**
+ * @brief The key of the Job ID.
+ */
+#define JOB_ID_KEY                      "jobId"
+#define JOB_ID_KEY_LENGTH               ( sizeof( JOB_ID_KEY ) - 1 )
+
+/**
+ * @brief Keys for the Job JSON response.
+ */
+#define JOB_DOC_KEY                     "jobDocument"
+#define JOB_DOC_KEY_LENGTH              ( sizeof( JOB_DOC_KEY ) - 1 )
+
+#define JOB_ACTION_KEY                  "action"
+#define JOB_ACTION_KEY_LENGTH           ( sizeof( JOB_ACTION_KEY ) - 1 )
+
+#define JOB_MESSAGE_KEY                 "message"
+#define JOB_MESSAGE_KEY_LENGTH          ( sizeof( JOB_MESSAGE_KEY ) - 1 )
+
+#define JOB_TOPIC_KEY                   "topic"
+#define JOB_TOPIC_KEY_LENGTH            ( sizeof( JOB_TOPIC_KEY ) - 1 )
+
+/**
+ * @brief Max Lengths for JSON values.
+ */
+#define JOBS_DEMO_MAX_ID_LENGTH         64
+#define JOBS_DEMO_MAX_JOB_DOC_LENGTH    128
+
+/**
+ * @brief The previous JobId and length
+ */
+static char _pJobId[ JOBS_DEMO_MAX_ID_LENGTH + 1 ] = { 0 };
+static size_t _JobIdLength = 0;
+
+/**
+ * @brief The Job Document which will be passed into the Notify Next callback.
+ */
+static char _jobDoc[ JOBS_DEMO_MAX_JOB_DOC_LENGTH + 1 ] = { 0 };
+static size_t _jobDocLength = 0;
+
+/**
+ * @brief Execute the Jobs Demo.
+ *
+ * This is the top level function called by the IOT Demo harness.
+ */
+int RunJobsDemo( bool awsIotMqttMode,
+                 const char * pIdentifier,
+                 void * pNetworKServerInfo,
+                 void * pNetworkCredentialInfo,
+                 const IotNetworkInterface_t * pNetworkInterface );
+
+/**
+ * @brief Initialize The Jobs Demo.
+ *
+ * Initialize MQTT module then the Jobs module. If the Jobs module fails
+ * to initialize, MQTT cleanup is invoked.
+ *
+ * @return EXIT_SUCCESS if initialization succeeds. Otherwise EXIT_FAILURE.
+ */
+static int32_t _initDemo( void )
```

Comment:
This function returns EXIT_SUCCESS/EXIT_FAILURE, so its return value should be `int`.

(Also other places where EXIT_SUCCESS/EXIT_FAILURE should be int)